boa/vyper/contract.py

```diff
@@ -497,15 +497,18 @@ def ast_map(self):
 
     def _get_fn_from_computation(self, computation):
         node = self.find_source_of(computation.code)
-        if isinstance(node, vy_ast.FunctionDef):
-            return node
-        return node.get_ancestor(vy_ast.FunctionDef)
+        fn_node = get_fn_ancestor_from_node(node)
+        if fn_node:
```

Comment:
this branch is unnecessary, since if it is not hit, `None` will be returned anyway